servers/dsub/jobs/test/base_test_cases.py

```diff
@@ -438,3 +454,146 @@ def test_query_jobs_submission_pagination(self):
                     page_token=response.next_page_token,
                     extensions=ExtendedQueryFields(submission=min_time)),
                 [job2])
+
+        def assert_status_counts_equal(self, counts1, counts2):
+            counts1Dict = {}
+            for status_count in counts1.counts:
+                counts1Dict[status_count.status] = status_count.count
+
+            counts2Dict = {}
+            for status_count in counts2.counts:
+                counts2Dict[status_count.status] = status_count.count
+
+            return self.assertEqual(counts1Dict, counts2Dict)
+
+        def test_get_job_aggregations(self):
+            name = flask.current_app.config['AGGREGATION_JOB_NAME_FILTER']
+
+            labels = {
+                'aggregation-testing-unique-1': 'testing-rocks',
+                'aggregation-testing-unique-2': 'debugging-bad'
+            }
+
+            other_values_labels = {
+                'aggregation-testing-unique-1': 'different-value',
+                'aggregation-testing-unique-2': 'also-different-value'
+            }
+
+            different_labels = {'different-label': 'different-value'}
+
+            labels_no_show = {'no-show': 'cannot-see-me'}
+
+            # Create an aborted job
+            job_aborted = self.api_job_id(
+                self.start_job('sleep 30', labels=labels, name=name))
+            time.sleep(10)
+            self.wait_status(job_aborted, ApiStatus.RUNNING)
+            self.must_abort_job(job_aborted)
+            self.wait_status(job_aborted, ApiStatus.ABORTED)
+
+            # Create a failed job
+            job_failed = self.api_job_id(
+                self.start_job('not_a_command', labels=labels, name=name))
+            self.wait_status(job_failed, ApiStatus.FAILED)
+
+            # Created a succeeded job
+            job_failed = self.api_job_id(
+                self.start_job('echo SUCCEEDED', labels=labels, name=name))
+            self.wait_status(job_failed, ApiStatus.SUCCEEDED)
+
+            # Created a running job
+            job_running = self.api_job_id(
+                self.start_job('sleep 30', labels=labels, name=name))
+            self.wait_status(job_running, ApiStatus.RUNNING)
```

Comment:
I would start all of these jobs first, then wait for all of them.

You should also bump up the sleep higher than, probably more like 180. I could imagine it taking >30s before you actually call the aggregations.

Ideally you also try to cancel all of these after the test runs (you can ignore failures there).